storage/aptosdb/src/utils/iterators.rs

```diff
@@ -285,3 +292,112 @@ impl<'a> Iterator for EventsByVersionIter<'a> {
+}
+
+impl<'a> AccountTransactionSummariesIter<'a> {
+    pub fn new(
+        inner: SchemaIterator<'a, TransactionSummariesByAccountSchema>,
+        address: AccountAddress,
+        start_version: Option<Version>,
+        end_version: Option<Version>,
+        limit: u64,
+        direction: ScanDirection,
+        ledger_version: Version,
+    ) -> Self {
+        Self {
+            inner,
+            address,
+            start_version,
+            end_version,
+            limit,
+            direction,
+            ledger_version,
+            prev_version: None,
+            count: 0,
+        }
+    }
+}
+
+impl<'a> AccountTransactionSummariesIter<'a> {
+    fn next_impl(&mut self) -> Result<Option<(Version, IndexedTransactionSummary)>> {
+        Ok(match self.inner.next().transpose()? {
+            Some(((address, version), txn_summary)) => {
+                // No more transactions sent by this account.
+                if address != self.address {
+                    return Ok(None);
+                }
+
+                // If already iterated over `limit` transactions, return None.
+                if self.count >= self.limit {
+                    return Ok(None);
+                }
+
+                // This case ideally shouldn't occur if the iterator is initiated properly.
+                if (self.direction == ScanDirection::Backward
+                    && version > self.end_version.unwrap())
+                    || (self.direction == ScanDirection::Forward
+                        && version < self.start_version.unwrap())
+                {
+                    return Ok(None);
+                }
+
+                ensure!(
+                    version == txn_summary.version(),
+                    "DB corruption: version mismatch: version in key: {}, version in txn summary: {}",
+                    version,
+                    txn_summary.version(),
+                );
+
+                // Ensure version_{i+1} > version_{i}
+                if let Some(prev_version) = self.prev_version {
+                    if self.direction == ScanDirection::Forward {
+                        ensure!(
+                            prev_version < version,
+                            "DB corruption: account transaction versions are not strictly increasing when scanning forward: \
+                             previous version: {}, current version: {}",
+                            prev_version,
+                            version,
+                        );
+                    } else {
+                        ensure!(
+                            prev_version > version,
+                            "DB corruption: account transaction versions are not strictly decreasing when scanning backward: \
+                             previous version: {}, current version: {}",
+                            prev_version,
+                            version,
+                        );
+                    }
+                }
```

Comment:
nit: this is a bit too paranoid, I would remove.

Reply:
Done.